diff --git a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs
--- a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs
+++ b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs
@@ -55,11 +55,11 @@
         if (opts.optimize) {
           plugins.push(adjustStaticMembers, adjustTypeScriptEnums, elideAngularMetadata);

-          // If the current file is denoted as explicit side effect free, add the pure
-          // top-level functions optimization plugin for this file.
-          if (opts.optimize.isSideEffectFree && opts.optimize.isSideEffectFree(args.path)) {
-            plugins.push(markTopLevelPure);
-          }
+          // For TensorBoard: This plugin aggressively culls symbols in a way that
+          // is incompatible with TensorBoard source. Disable it. As result the binary is bigger.
+          //if (opts.optimize.isSideEffectFree && opts.optimize.isSideEffectFree(args.path)) {
+          //  plugins.push(markTopLevelPure);
+          //}
         }

         const shouldRunLinker =
